### PR TITLE
Add methods to get, accept and decline remote shares

### PR DESCRIPTION
**Pull request by nickvergessen**
@PVince81 

Requires: 
- [x] https://github.com/owncloud/core/pull/16733

### CODE REVIEW

**Comment by nickvergessen** [comment]
The core PR is in, I rebased this PR on current master.

Would be nice to get this in soon ;)

**Comment by PVince81** [comment]
I guess there is no good way to write a test for this ?

Did you test what happens when the API is not available, for example in 8.0.5 ?

**Comment by nickvergessen** [comment]
Status: 999
```xml
<message>Invalid query, please check the syntax. API specifications are here: http://www.freedesktop.org/wiki/Specifications/open-collaboration-services. DEBUG OUTPUT:
debug output:
http request method: GET
http request uri: /ownCloud/Smashbox/master/ocs/v1.php/apps/files_sharing/api/v1/remote_shares
</message>
```

**Comment by PVince81** [comment]
Hmm yeah. At some point we'll want to map 999 to a custom exception.

:+1: for now

File: owncloud/owncloud.py [hunk 1]
```diff
@@ -831,7 +898,7 @@ def share_file_with_user(self, path, user, **kwargs):
 
         path = self.__normalize_path(path)
         post_data = {
-            'shareType': self.OCS_SHARE_TYPE_USER,
+            'shareType': self.OCS_SHARE_TYPE_REMOTE if remote_user else self.OCS_SHARE_TYPE_USER,
```
Comment (PVince81) [inline owncloud/owncloud.py]: someone had fun with python syntax ? :wink:

Comment (nickvergessen) [inline owncloud/owncloud.py]: yeah, took me a while :see_no_evil:

**Pull request merged**